File: week-2/subdomain-visit-count/subdomain-visit-count.js
/**
 * @param {string[]} cpdomains
 * @return {string[]}
 */
var subdomainVisits = function(cpdomains) {
    const hashtable = {};
    for (let domain of cpdomains) {
        let [count, string] = domain.split(' ');
        count = parseInt(count);
        const array = string.split('.');
        for (let i=0; i<array.length; i++) {
            const string = array.slice(i).join('.');
            if (string in hashtable)
                hashtable[string] = hashtable[string] + count;
            else
                hashtable[string] = count;
        }
    }
    
    return Object.entries(hashtable).map(array => `${array[1]} ${array[0]}`);
    
};

File: week-2/subdomain-visit-count/subdomain-visit-count.py
class Solution:
    def subdomainVisits(self, cpdomains: List[str]) -> List[str]:
        hashtable = {}
        for domain in cpdomains:
            count, string = domain.split(' ')
            count = int(count)
            array = string.split('.')
            for i in range(len(array)):
                string = '.'.join(array[i:])
                hashtable[string] = hashtable[string] + count if string in hashtable else count
                
        return [str(val) + ' ' + key for key, val in hashtable.items()]
